swift/Sources/FlatBuffers/FlatBufferBuilder.swift

```diff
@@ -14,7 +14,11 @@
  * limitations under the License.
  */
 
+#if !os(WASI)
 import Foundation
+#endif
+
+import SwiftOverlayShims
 
```

Comment:
I'm not sure if Foundation can be imported on linux + Can we not import SwiftOverlayShims, unless we are actually using `WASI`? 

similar to what you did in `ByteBuffer.swift`